### PR TITLE
WS-94 - NFT class

### PR DESCRIPTION
**Pull request by AgusVelez5**
**Type of MR:**

- [x] Feature
- [ ] Fix
- [ ] HotFix

**Small description:**

  - Add NFT class

**Aditional observations/considerations:**
...

# Checklist:

- [x] My code follows the styleguides of the project (successfully pass linters)
- [x] I tested my code.
- [x] I commented/docummented my code in a way that complex parts are explained.
- [ ] If required, i updated project's README file.
- [x] I added/updated tests that tests my changes in code.
- [x] All previous and new tests pass.
- [ ] I commit any new dependency.
- [ ] I updated UML diagram in confluence if any model was modified in this MR.

### CODE REVIEW

File: src/types/interfaces.ts [hunk 1]
```diff
@@ -116,3 +116,12 @@ export interface DaiPermitMessage {
   expiry: BigNumberish
   allowed?: boolean
 }
+
+export interface NftMetadata {
+  name?: string
+  description?: string
+  image?: string
+  traits?: Array<object>
```
Comment (mauricio-b-r) [inline src/types/interfaces.ts:124]: Is the traits property needed? What for?

File: src/nft/NFT.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    return this._jsonData
+  }
+  // end getters
+
+  /**
+   * saves image base64 on image attribute
+   * @param  {Ipfs} ipfs ipfs instance.
+   * @return {Promise<void>}
+   */
+  async saveImageBase64(ipfs: Ipfs = isRequired('ipfs')) {
```
Comment (mauricio-b-r) [inline src/nft/NFT.ts:66]: I don't think this is a good name for the function, also what if the image is not stored on ipfs?

File: src/types/interfaces.ts (hunk 1 above)
Comment (mauricio-b-r) [reply src/types/interfaces.ts:124]: Also, both EIPs 721 and 1155 tend to use a properties objects
`"properties": {
        "name": {
            "type": "string",
            "description": "Identifies the asset to which this token represents",
        }
    }`

File: src/nft/NFT.ts (hunk 2 above)
Comment (fedeaviles) [reply src/nft/NFT.ts:67]: You should use the new interface `StorageType`

Comment (AgusVelez5) [reply src/nft/NFT.ts:67]: I thought about that, but IPFSStorage and AWSStorage are not available yet. So I thought to send it with the current state and then refactor it in a new task
any recommendations for the name?

Comment (mauricio-b-r) [reply src/nft/NFT.ts:67]: I think that the StorageType interface is already on develop so you should use that. And for the name I think it should be something more like "fetchBase64Image"

File: src/types/interfaces.ts (hunk 1 above)
Comment (mauricio-b-r) [reply src/types/interfaces.ts:124]: mm but on the EIPs they are defined like that: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md, https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1155.md#erc-1155-metadata-uri-json-schema

File: src/nft/NFT.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  get jsonData(): NftMetadata {
+    return this._jsonData
+  }
+  // end getters
+
+  /**
+   * saves image base64 on image attribute
+   * @param  {Ipfs} ipfs ipfs instance.
```
Comment (mauricio-b-r) [inline src/nft/NFT.ts:64]: update docstring

**Review by mauricio-b-r — APPROVED** [review]

**Review by mauricio-b-r — APPROVED** [review]

**Pull request merged**